### PR TITLE
Helm unit tests

### PR DESCRIPTION
**Pull request by pdabelf5**
### Proposed changes

Add Helm template unit tests

### Testing
Run tests
```
make test
```

Update snapshots
```
make test-update-snaps
```
### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by j1m-ryan — APPROVED** [review]

File: charts/tests/helmunit_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+			expectedErr: nil,
+			releaseName: "appprotect-dos",
+			namespace:   "appprotect-dos",
+		},
+	}
+
+	// Path to the helm chart we will test
+	helmChartPath, err := filepath.Abs("../nginx-ingress")
+	if err != nil {
+		t.Error("Failed to open helm chart path ../nginx-ingress")
```
Comment (jjngx) [inline charts/tests/helmunit_test.go:99]: `t.Fatal` would fail the test immediately in case there is an error opening the file.

File: go.mod [hunk 2]
```diff
@@ -32,7 +33,9 @@ require (
 
 require (
 	github.com/Azure/go-ntlmssp v0.0.0-20221128193559-754e69321358 // indirect
+	github.com/BurntSushi/toml v1.3.2 // indirect
```
Comment (vepatel) [inline go.mod:36]: @jjngx @pdabelf5 do we need to get approval for new libs?

**Comment by vepatel** [comment]
@pdabelf5 can we exclude tests/ from operator sync job?

File: go.mod (hunk 2 above)
Comment (pdabelf5) [reply go.mod:36]: The only direct dependency added is `github.com/gruntwork-io/terratest v0.47.2`

Comment (jjngx) [reply go.mod:36]: @vepatel both dependencies have MIT license. It should be straightforward.

**Review by jjngx — APPROVED** [review]

File: charts/tests/helmunit_test.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+// An example of how to verify the rendered template object of a Helm Chart given various inputs.
+func TestHelmNICTemplate(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		valuesFile  string
+		releaseName string
+		namespace   string
+		expectedErr error
```
Comment (jjngx) [inline charts/tests/helmunit_test.go:32]: is the `expectedErr` field used in the tests?

File: go.mod (hunk 2 above)
Comment (pdabelf5) [reply go.mod:36]: Thanks @jjngx

File: charts/tests/helmunit_test.go (hunk 3 above)
Comment (pdabelf5) [reply charts/tests/helmunit_test.go]: No, I will remove it now

**Comment by pdabelf5** [comment]
> @pdabelf5 can we exclude tests/ from operator sync job?

The operator syncs the released chart, not the directory from NIC

File: go.mod (hunk 2 above)
Comment (pdabelf5) [reply go.mod:36]: `github.com/gruntwork-io/terratest` has a pre-approved Apache 2.0 license

**Review by vepatel — APPROVED** [review]

**Review by haywoodsh — APPROVED** [review]

**Pull request merged**